src/psyclone/domain/gocean/transformations/raise_psyir_2_gocean_kern_trans.py

```diff
@@ -0,0 +1,217 @@
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author R. W. Ford STFC Daresbury Lab
+
+'''Raise generic PSyIR representing a kernel-layer routine to
+PSyclone kernel-layer-specific PSyIR which uses specialised classes.
+
+'''
+import re
+
+from psyclone.domain.gocean.kernel import GOceanKernelMetadata, GOceanContainer
+from psyclone.psyGen import Transformation
+from psyclone.psyir.nodes import Container, ScopingNode, FileContainer
+from psyclone.psyir.transformations import TransformationError
+
+
+def find_symbol(node, name):
+    '''Utility method to find the symbol associated with the supplied
+    name. The supplied node and all of its siblings are searched and the
+    symbol and its scoping node are returned if the symbol is found.
+
+    :param node: the starting node for the search.
+    :type: :py:class:`psyclone.psyir.nodes.node`
+    :param str name: the name of the symbol being searched.
+
+    :returns: a tuple containing the symbol with the same name as the \
+        supplied name and the scoping node for that symbol.
+    :rtype: Optional[(:py:class:`psyclone.psyir.symbols.datasymbol, \
+        :py:class:`psyclone.psyir.nodes.node`)]
+
+    '''
+    symbol = None
+    scoping_node = None
+    for test_node in node.walk(ScopingNode):
+        if name in test_node.symbol_table:
+            symbol = test_node.symbol_table.lookup(name)
+            scoping_node = test_node
+            break
+    return (symbol, scoping_node)
+
+
+class RaisePSyIR2GOceanKernTrans(Transformation):
+    '''Raise a generic PSyIR representation of a kernel-layer routine
+    to a PSyclone version with specialised domain-specific nodes and
+    symbols. This is currently limited to the specialisation of kernel
+    metadata.
+
+    >>> from psyclone.domain.gocean.transformations import kern_trans
+    >>> from psyclone.psyir.frontend.fortran import FortranReader
+    >>> CODE = (
+    ...     "MODULE example\n"
```

Comment:
The formatting here is giving a lot of errors when building the reference guide (I think you need to remove the indentation of each string).

Reply:
Hopefully fixed now. That took longer than I would have liked.